server/message_hooks_test.go

```diff
@@ -2880,3 +2881,136 @@ func TestUserWillLogin(t *testing.T) {
 		})
 	}
 }
+
+func TestMessageHasBeenPosted(t *testing.T) {
+	assert := require.New(t)
+	th := setupTestHelper(t)
+
+	// TODO: remove this when the issue is fixed in the API
+	getDirectChannel := func(t *testing.T, senderID, receiverID string) *model.Channel {
+		t.Helper()
+
+		channel, appErr := th.p.API.GetDirectChannel(senderID, receiverID)
+		if appErr != nil {
+			if appErr.Id != "app.sharedchannel.dm_channel_creation.internal_error" {
+				assert.Nil(appErr)
+			}
+			channel, _ = th.p.API.GetDirectChannel(senderID, receiverID)
+		}
+		return channel
+	}
+
+	setup := func(t *testing.T, selectiveSync bool) (msTeamsUser, MMUser, syntheticUser *model.User) {
+		th.ResetPreserveDB(t)
+
+		team := th.SetupTeam(t)
+		msTeamsUser = th.SetupUser(t, team)
```

Comment:
nit: I thought `msTeamsUser` would be an instance of the remote Teams structure. Should this be named `teamsPrimaryUser` (and similarly `mmPrimaryUser`?)

Reply:
Good call, changing the names for extra clarity.